web/packages/teleterm/src/ui/services/workspacesService/accessRequestsService.ts

```diff
@@ -258,6 +329,7 @@ export type ResourceRequest =
       kind: 'kube';
       resource: {
         uri: KubeUri;
+        namespaces?: Map<ResourceUri, ResourceRequest>;
```

Comment:
```suggestion
        namespaces?: Map<KubeResourceNamespaceUri, ResourceRequest>;
```

Also, does `ResourceRequest` need to be recursive? We accept `kind: 'namespace'` both within the overall type (`ResourceRequest`) and we also accept it here under the `namespaces` map. Is it necessary? Can we make it so that `kind: 'namespace'` is not within the overall type and is accepted only within a kube resource?

If the answer to this is false, then I still think it'd make sense to extract a private type like so:

```typescript
export type ResourceRequest =
  // …
  | {
      kind: 'kube';
      resource: {
        uri: KubeUri;
        namespaces?: Map<KubeResourceNamespaceUri, NamespaceResourceRequest>;
      };
    }
  // …
  | NamespaceResourceRequest;

type NamespaceResourceRequest = {
  kind: 'namespace';
  resource: {
    uri: KubeResourceNamespaceUri;
  };
};

```

Reply:
i think this comments is resolved with @gzdunek code: https://github.com/gravitational/teleport/pull/47347#discussion_r1806718255